### PR TITLE
[X86][CodeGen] Support flags copy lowering for NDD ADC/SBB/RCL/RCR

### PR DESCRIPTION
**Pull request by KanRobert**

### CODE REVIEW

File: llvm/lib/Target/X86/X86FlagsCopyLowering.cpp [hunk 1]
```diff
@@ -806,7 +811,8 @@ void X86FlagsCopyLoweringPass::rewriteArithmetic(
   // Insert an instruction that will set the flag back to the desired value.
   Register TmpReg = MRI->createVirtualRegister(PromoteRC);
   auto AddI =
-      BuildMI(MBB, MI.getIterator(), MI.getDebugLoc(), TII->get(X86::ADD8ri))
+      BuildMI(MBB, MI.getIterator(), MI.getDebugLoc(),
+              TII->get(Subtarget->hasNDD() ? X86::ADD8ri_ND : X86::ADD8ri))
```
Comment (XinWang10) [inline llvm/lib/Target/X86/X86FlagsCopyLowering.cpp:815]: Do we need to use macro like GET_NDD_IF_ENABLED(OPC_ND, OPC).

**Review by XinWang10 — APPROVED** [review]
LGTM with 1 comment

File: llvm/lib/Target/X86/X86FlagsCopyLowering.cpp (hunk 1 above)
Comment (KanRobert) [reply llvm/lib/Target/X86/X86FlagsCopyLowering.cpp:815]: We may need to define it in other files. But in this file, only one use.

**Comment by KanRobert** [comment]
Thanks @XinWang10 !

**Pull request merged**